File: unit-3/sprint-3/day-3/you/MyMusicApp/app/src/main/java/com/example/apmusicapp/MusicAdapter.java
package com.example.apmusicapp;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MusicAdapter extends RecyclerView.Adapter<MusicViewHolder> {
    ArrayList<MusicModel> musicModels;
    private OnSongClicked onSongClicked;

    public MusicAdapter(ArrayList<MusicModel> musicModels, OnSongClicked onSongClicked) {
        this.musicModels = musicModels;
        this.onSongClicked = onSongClicked;
    }

    @NonNull
    @Override
    public MusicViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout, parent, false);
        return new MusicViewHolder(view, onSongClicked);
    }

    @Override
    public void onBindViewHolder(@NonNull MusicViewHolder holder, int position) {
        MusicModel musicModel = musicModels.get(position);
        holder.setData(musicModel);
    }

    @Override
    public int getItemCount() {
        return musicModels.size();
    }
}

File: placements_assignment/musicapp/app/src/main/java/dev/amol/musicapp/ui/adapter/MusicAdapter.kt
package dev.amol.musicapp.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import dev.amol.musicapp.R
import dev.amol.musicapp.databinding.ItemlayoutBinding
import dev.amol.musicapp.model.remote.responsedata.Result
import kotlin.coroutines.coroutineContext

class MusicAdapter(
     val context: Context,
    private val musicList: MutableList<Result>
):RecyclerView.Adapter<MusicAdapter.MusicViewHolder>() {

    class MusicViewHolder(val itemlayoutBinding: ItemlayoutBinding):RecyclerView.ViewHolder(itemlayoutBinding.root){
        fun setData(result: Result){
            itemlayoutBinding.musicData = result

            Glide.with(itemlayoutBinding.ivArtistImg).load(result.artworkUrl100).into(itemlayoutBinding.ivArtistImg)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MusicViewHolder {
        return MusicViewHolder(DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.itemlayout,parent,false))
    }

    override fun onBindViewHolder(holder: MusicViewHolder, position: Int) {
        val result = musicList[position]
        holder.setData(result)
    }

    override fun getItemCount(): Int {
        return musicList.size
    }
}